File: go/fixture/go-mini/internal/mocks/notifier.go
package mocks

// Notifier interface avoids an import cycle with the alerts package
type NotifierAPI interface {
	Send(channel, msg string) error
}

// Notifier records every message sent through it.
type Notifier struct {
	Sent []string
}

var _ NotifierAPI = (*Notifier)(nil)

// Send records the message.
func (n *Notifier) Send(channel, msg string) error {
	n.Sent = append(n.Sent, channel+": "+msg)
	return nil
}

File: py/fixture/py-mini/internal/mocks/notifier.py
"""A notifier double that records what it was asked to send."""

from typing import Protocol


class NotifierAPI(Protocol):
    """Notifier interface; avoids an import cycle with the alerts package."""

    def send(self, channel: str, msg: str) -> None:
        """Send a message on a channel."""
        ...


class Notifier:
    """Records every message sent through it."""

    def __init__(self) -> None:
        self.sent: list[str] = []

    def send(self, channel: str, msg: str) -> None:
        """Record the message."""
        self.sent.append(channel + ": " + msg)


_double: type[NotifierAPI] = Notifier
